src/irr/asset/CCPUMeshPacker.h

```diff
@@ -143,36 +155,155 @@ ReservedAllocationMeshBuffers CCPUMeshPacker<MDIStructType>::alloc(const Iterato
 }
 
 template <typename MDIStructType>
-void CCPUMeshPacker<MDIStructType>::commit()
+void CCPUMeshPacker<MDIStructType>::createOutputBuffers()
 {
-	//for (int i = 0; i < MDIStructsNeeded; i++)
-	//{
-	//	uint32_t idxCnt = 0u;
-
-	//	if (i == MDIStructsNeeded - 1)
-	//	{
-	//		idxCnt = static_cast<uint32_t>(meshBuffer->getIndexCount()) % maxIdxCntPerPatch;
-
-	//		//TODO: test for this case
-	//		if (idxCnt == 0)
-	//			idxCnt = maxIdxCntPerPatch;
-	//	}
-	//	else
-	//	{
-	//		idxCnt = maxIdxCntPerPatch;
-	//	}
-
-	//	*(mdiBuffPtr + MDIAllocAddr + i) =
-	//	{
-	//		idxCnt,
-	//		static_cast<uint32_t>(meshBuffer->getInstanceCount()),
-	//		idxAllocAddr + i * maxIdxCntPerPatch,
-	//		0 /*TODO*/,
-	//		static_cast<uint32_t>(meshBuffer->getBaseInstance()) /*TODO*/
-	//	};
-	//}
-
-	return;
+	if (outputBuffer.indexBuffer.buffer != nullptr)
+		return;
+
+	//TODO: redo after safe_shrink fix
+	outputBuffer.MDIDataBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(m_allocParams.MDIDataBuffSupportedSizeInBytes);
+	outputBuffer.indexBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(m_allocParams.indexBuffSupportedSizeInBytes);
+	outputBuffer.perVertexVtxBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(m_allocParams.vertexBuffSupportedSizeInBytes);
+	if(m_perInstVtxSize)
+		outputBuffer.perInstanceVtxBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(m_allocParams.perInstanceVertexBuffSupportedSizeInBytes);
```

Comment:
you can use `address_allocator_traits` or the allocator itself and query the total size of its address space, then multiply that by the `sizeof` the units its working in and you'll get the buffer size you need.